### DISCUSSION TITLE
Where do you get `navigation` from?

### DISCUSSION THREAD

**Discussion by trajano** [pendo-io/pendo-mobile-sdk#80] [Q&A] [answered]
In the [React Native Navigation guide](https://github.com/pendo-io/pendo-mobile-sdk/blob/master/ios/pnddocs/expo_rnn.md) there's a reference to `navigation`.  Where is that obtained and what is supposed to be the type because it's currently marked as `any`.

**Comment by noambartouv** [comment, marked as answer]
Hey @trajano, we just refreshed our documenation, including API documentation. I believe you can find you answer here: [RN NavigationOptions](https://github.com/pendo-io/pendo-mobile-sdk/blob/master/api-documentation/rn-apis.md#navigationoptions)

The `navigation` object is only required when using the React Native Navigation library. If you are using that library we require passing in the RNN navigation instance. Let us know if you have further questions.

**Reply by trajano** [reply]
It says...

> The Navigation object when integrating the React Native Navigation SDK. Otherwise ignore this parameter

But the way you worded it in the answer seems grammatically better.  Though I would tweak it a bit with emphasis as follows since it's easy to miss

> The `navigation` property is  required **if and only if** the React Native Navigation library is used for navigation. For example:
> ```typescript
> import { Navigation } from "react-native-navigation";
> ...
> const navigationOptions = { library: NavigationLibraryType.ReactNativeNavigation, navigation: Navigation };
> PendoSDK.setup(pendoKey, navigationOptions);
> ```
> If you are using another library such as React Navigation for navigation, the value does not need to be specified for example. 
> ```typescript
> const navigationOptions = { library: NavigationLibraryType.ReactNavigation };
> PendoSDK.setup(pendoKey, navigationOptions);
> ```

**Reply by noambartouv** [reply]
@trajano thank you for the feedback. I've updated the descreption and examples accordingly.

**Reply by trajano** [reply]
@noambartouv 
you had a typo.
It's `React Navigation` not `React Native` on the first example

**Reply by noambartouv** [reply]
@trajano Updated. Thank you for pointing it out.
